required/latex-lab/latex-lab-toc-kernel-changes.dtx

```diff
@@ -227,17 +222,27 @@
 % The text and the page
 % hooks are used in manual \cs{l@xx} commands and in \cs{l@xx} defined by
 % \cs{@dottedtocline} or by other means.
+% The arguments of the following hooks are:
+%  \begin{itemize}
+%  \item[\#1] level as a \emph{number} (integer expression), see above for the standard
+%   numbering. The standard first argument of \cs{contentsline} is not used, as
+%   \cs{@dottedtocline} knows only a number.
+%  \item[\#2] content (the second argument of a \cs{contentsline}
+%  \item[\#3] page    (the third argument of a \cs{contentsline}
+%  \item[\#3] target  (the fourth argument of a \cs{contentsline}
+%  \end{itemize}
 %    \begin{macrocode}
-\NewMirroredHookPair{contentsline/number/before}{contentsline/number/after}%
-\NewMirroredHookPair{contentsline/text/before}{contentsline/text/after}%
-\NewMirroredHookPair{contentsline/page/before}{contentsline/page/after}%
+\NewMirroredHookPairWithArguments{contentsline/number/before}{contentsline/number/after}{1}%
+\NewMirroredHookPairWithArguments{contentsline/text/before}{contentsline/text/after}{4}%
+\NewMirroredHookPairWithArguments{contentsline/page/before}{contentsline/page/after}{4}%
 %    \end{macrocode}
```

Comment:
TODO: convince myself if we really need all of these hooks, I doubt it. In particular I doubt that the /after hooks need to have arguments (or need to be there).

Reply:
well I need to inject link + tagging code around the number, the text and the page. And this code must be mirrored: the link code must be either inside or outside the tagging code. I don't quite see how  do it differently or how to produce a pair where the begin takes arguments and the end hook not.

Reply:
make the number an argument to the hook maybe, really comes down to configuration points by the end of the day ... but it isn't something that need to be solved or changed now, but perhaps add a TODO to review the current approach